public/app/plugins/panel/singlestat/module.ts

```diff
@@ -378,19 +310,62 @@ class SingleStatCtrl extends MetricsPanelCtrl {
     this.panel.rangeMaps.push({ from: '', to: '', text: '' });
   }
 
+  /**
```

Comment:
We could move this process to `handleDataFrame(...)`, but would need to update the editor to call `refresh()` on any settings change rather than `render()` -- I made that change for enabling sparklines already